lib/plugins/aws/deploy/lib/uploadArtifacts.js

```diff
@@ -83,26 +83,32 @@ module.exports = {
 
     const functionNames = this.serverless.service.getAllFunctions();
     let artifactFilePaths = _.uniq(
-      functionNames.map(name => {
-        const functionArtifactFileName = this.provider.naming.getFunctionArtifactName(name);
-        const functionObject = this.serverless.service.getFunction(name);
-        functionObject.package = functionObject.package || {};
-        const artifactFilePath =
-          functionObject.package.artifact || this.serverless.service.package.artifact;
-
-        if (
-          !artifactFilePath ||
-          (this.serverless.service.artifact && !functionObject.package.artifact)
-        ) {
-          if (this.serverless.service.package.individually || functionObject.package.individually) {
-            const artifactFileName = functionArtifactFileName;
-            return path.join(this.packagePath, artifactFileName);
+      functionNames
+        .map(name => {
+          const functionObject = this.serverless.service.getFunction(name);
+          if (functionObject.image) return null;
+          const functionArtifactFileName = this.provider.naming.getFunctionArtifactName(name);
+          functionObject.package = functionObject.package || {};
+          const artifactFilePath =
+            functionObject.package.artifact || this.serverless.service.package.artifact;
+
+          if (
+            !artifactFilePath ||
+            (this.serverless.service.artifact && !functionObject.package.artifact)
+          ) {
+            if (
+              this.serverless.service.package.individually ||
+              functionObject.package.individually
+            ) {
+              const artifactFileName = functionArtifactFileName;
+              return path.join(this.packagePath, artifactFileName);
+            }
+            return path.join(this.packagePath, this.provider.naming.getServiceArtifactName());
           }
-          return path.join(this.packagePath, this.provider.naming.getServiceArtifactName());
-        }
 
-        return artifactFilePath;
-      })
+          return artifactFilePath;
+        })
+        .filter(Boolean)
```

Comment:
TIL that this will filter nulls.

Reply:
It filters all falsy values, which is fine for collection of objects or strings (were we do not envision empty string as valid).

Still it can bite if we would filter collection of numbers, as `0` is falsy

Reply:
Good to know, thanks!